src/diffpy/utils/scattering_objects/diffraction_objects.py

```diff
@@ -15,12 +15,38 @@ def __init__(self, name='', wavelength=None):
         self.name = name
         self.wavelength = wavelength
         self.scat_quantity = ""
-        self.on_q = [np.empty(1), np.empty(1)]
-        self.on_tth = [np.empty(1), np.empty(1)]
-        self.on_d = [np.empty(1), np.empty(1)]
+        self.on_q = [np.empty(0), np.empty(0)]
+        self.on_tth = [np.empty(0), np.empty(0)]
+        self.on_d = [np.empty(0), np.empty(0)]
         self._all_arrays = [self.on_q, self.on_tth]
         self.metadata = {}
 
+    def __eq__(self, other):
+        if not isinstance(other, Diffraction_object):
+            return NotImplemented
+
+        attributes_to_test = {"name", "wavelength", "scat_quantity", "on_q", "on_tth", "on_d", "metadata"}
```

Comment:
since you define attributes to test here, why not import it into the test function, or vice versa?   Have a think if that makes sense.

Reply:
I agree. I think it probably makes more sense to define it as a global parameter in the function and then import it to test function?